lib/X25519KeyAgreementKey2020.js

```diff
@@ -0,0 +1,375 @@
+        'Export requires specifying either "publicKey" or "privateKey".');
+    }
+    const exportedKey = {
+      id: this.id,
+      type: this.type,
+      controller: this.controller
+    };
+    if(publicKey) {
+      exportedKey.publicKeyMultibase = this.publicKeyMultibase;
+    }
+    if(privateKey) {
+      exportedKey.privateKeyMultibase = this.privateKeyMultibase;
+    }
+    return exportedKey;
+  }
+
+  /**
+   * Generates and returns a multiformats encoded
+   * X25519 public key fingerprint (for use with cryptonyms, for example).
+   *
+   * @see https://github.com/multiformats/multicodec
+   *
+   * @param {object} [options={}] - Options hashmap.
+   * @param {string} options.publicKeyMultibase - Multibase encoded public key.
+   *
+   * @returns {string} The fingerprint.
+   */
+  static fingerprintFromPublicKey({publicKeyMultibase} = {}) {
+    if(!publicKeyMultibase) {
+      throw new Error('Source public key is required.');
+    }
+    // X25519 cryptonyms are multicodec encoded values, specifically:
+    // (multicodec('x25519-pub') + key bytes)
+    const pubkeyBytes = _multibaseDecode({text: publicKeyMultibase});
+    const buffer = new Uint8Array(2 + pubkeyBytes.length);
+    // See https://github.com/multiformats/multicodec/blob/master/table.csv
+    // 0xec is the value for X25519 public key
+    // 0x01 is from varint.encode(0xec) -> [0xec, 0x01]
+    // See https://github.com/multiformats/unsigned-varint
+    buffer[0] = 0xec; //
+    buffer[1] = 0x01;
+    buffer.set(pubkeyBytes, 2);
+    return _multibaseEncode({bytes: buffer});
+  }
+
+  /**
+   * Creates an instance of X25519KeyAgreementKey2020 from a key fingerprint.
+   *
+   * @param {object} [options={}] - Options hashmap.
+   * @param {string} options.fingerprint - Public key fingerprint.
+   *
+   * @returns {X25519KeyAgreementKey2020} Key pair instance (public key material
+   *   only) created from the fingerprint.
+   */
+  static fromFingerprint({fingerprint} = {}) {
+    if(!fingerprint ||
+      !(typeof fingerprint === 'string' && fingerprint[0] === 'z')) {
+      throw new Error('`fingerprint` must be a multibase encoded string.');
+    }
+    const buffer = _multibaseDecode({text: fingerprint});
+
+    // buffer is: 0xec 0x01 <public key bytes>
+    if(buffer[0] !== 0xec || buffer[1] !== 0x01) {
+      throw new Error(`Unsupported Fingerprint Type: ${fingerprint}`);
+    }
+
+    return new X25519KeyAgreementKey2020({
+      publicKeyMultibase: _multibaseEncode({bytes: buffer.slice(2)})
+    });
+  }
+
+  /**
+   * Derives a shared secret via a given public key, typically for use
+   * as one parameter for computing a shared key. It should not be used as
+   * a shared key itself, but rather input into a key derivation function (KDF)
+   * to produce a shared key.
+   *
+   * @param {object} [options={}] - Options hashmap.
+   * @param {LDKeyPair} options.publicKey - Remote key pair.
+   *
+   * @returns {string} Derived secret.
```

Comment:
I believe we are still trying to be explicit wrt async stuffs
```suggestion
   * @returns {Promise<string>} Derived secret.
```

Reply:
Oversight; fixed.